### ISSUE TITLE
upgrade to Microsoft.OpenApi 1.2.0 cause issue System.MissingMethodException: Method not found: 'Void Microsoft.OpenApi.Any.OpenApiString..ctor(System.String)'

### ISSUE THREAD

**Issue by topnguyen** [domaindrivendev/Swashbuckle#1378]
## STEPS TO REPRODUCE: ##
- Create new Asp Net Core 3.1 project
- Install Swashbuckle Libs
```xml
<PackageReference Include="Swashbuckle.AspNetCore" Version="5.4.1" />
<PackageReference Include="Swashbuckle.AspNetCore.Annotations" Version="5.4.1" />
```
- Install `Microsoft.OpenApi` version `1.2.0`

> You can download my sample project following the above steps [here](https://drive.google.com/file/d/13wXgWpm1RwU54KHxxxtBnR9Y12GGdyHQ/view?usp=sharing)

- Then access to the swagger doc and face the issue

![image](https://user-images.githubusercontent.com/15808788/84762770-6e43b200-aff5-11ea-8b88-7d58edfca8a2.png)

## EXPECTED RESULT: ##
Can access the swagger doc as well as run with `Microsoft.OpenApi` version `1.1.4` 

## ACTUAL RESULT: ##
```text
System.MissingMethodException: Method not found: 'Void Microsoft.OpenApi.Any.OpenApiString..ctor(System.String)'.
   at Swashbuckle.AspNetCore.SwaggerGen.OpenApiAnyFactory.CreateFor(OpenApiSchema schema, Object value)
```

**Comment by vpgalon16** [comment]
I encountered this problem when migrating to core 3.1. 
It failed to generate swagger.json, the workaround to revert back to version 1.1.4

**Comment by hasancemcerit** [comment]
This has been fixed with the merge request [#492](https://github.com/microsoft/OpenAPI.NET/pull/492), with version **1.2.2**

**Comment by topnguyen** [comment]
> This has been fixed with the merge request [#492](https://github.com/microsoft/OpenAPI.NET/pull/492), with version **1.2.2**

Have tried on version 1.2.2 and the issue resolved, thank for your notice @hasancemcerit
